MSBLOC.Core/Services/Parser.cs

```diff
@@ -12,6 +12,59 @@
 
 namespace MSBLOC.Core.Services
 {
+    public class SolutionDetails
+    {
+        private Dictionary<string, ProjectDetails> Projects { get; }
```

Comment:
Might be nice to have SolutionDetails inherit `Dictionary<string, ProjectDetails>` for all the IDictionary goodness.

For instance, with this implementation you'll get an exception if you try to get Project with a projectFile (key) that doesn't exist in th encapsalated Dictionary. Whereas you could test that easily with a free `ContainsKey` call with inheritance rather than encapsulation.

Reply:
👍